src/core/document.js

```diff
@@ -280,10 +287,23 @@ var Page = (function PageClosure() {
           return pageOpList;
         }
 
-        var annotationsReadyPromise = Annotation.appendToOperatorList(
-          annotations, pageOpList, partialEvaluator, task, intent,
-          renderInteractiveForms);
-        return annotationsReadyPromise.then(function () {
+        // Collect the operator list promises for the annotations. Each promise
+        // is resolved with the complete operator list for a single annotation.
+        var i, ii;
+        var opListPromises = [];
```

Comment:
Nit: I you want to offset the (tiny) increase in diff size from the previous comment, I think that you could simply do `var i, ii, opListPromises = [];` here instead since all these variables are sort of temporary anyway :-)